test/validation/api/queue/queue.c

```diff
@@ -83,7 +89,8 @@ static int queue_suite_init(void)
 	odp_pool_param_init(&params);
 
 	params.buf.size  = 4;
-	params.buf.align = ODP_CACHE_LINE_SIZE;
```

Comment:
Removal of this line would be the only change needed in this file. Adding code for reading pool capabilities etc seems unnecessary bloat here. There is nothing that actually relies on better buffer alignment than the default (and if there were, this patch would not handle it anyway).

I think the changes in the other files are ok even though there too better alignment than default is not really needed. The other changes are small and may improve performance (in the helpers). And in the pool validation case it is good that, at least in some cases, the alignment parameter is set.

Reply:
I would leave the changes in the helpers as they are now (i.e. cache line alignment if possible, default otherwise). Maybe it helps performance.

In time_global_test.c and queue.c I would just remove the whole alignment setting and rely on the default (which is at least 8), as I do not see any point requesting a larger alignment.

In pool.c setting the alignment might slightly increase test coverage, but since there are alignment tests in buffer and packet validation tests, I would just remove the explicit alignment setting from pool.c too.

Thus, the way I would do this would be to leave the helpers as they are in the latest patch and simply remove the alignment setting line from the other files so that the parameter gets set by odp_pool_param_init().

Reply:
Ack. I wil send the next version with above change